invokeai/app/services/model_install/model_install_base.py

```diff
@@ -0,0 +1,236 @@
+    ) -> str:
+        """
+        Probe, register and install the model in the models directory.
+
+        This moves the model from its current location into
+        the models directory handled by InvokeAI.
+
+        :param model_path: Filesystem Path to the model.
+        :param config: Dict of attributes that will override autoassigned values.
+        :returns id: The string ID of the registered model.
+        """
+
+    @abstractmethod
+    def import_model(
+        self,
+        source: Union[str, Path, AnyHttpUrl],
+        inplace: bool = False,
+        variant: Optional[str] = None,
+        subfolder: Optional[str] = None,
+        config: Optional[Dict[str, Any]] = None,
+        access_token: Optional[str] = None,
+    ) -> ModelInstallJob:
+        """Install the indicated model.
+
+        :param source: Either a URL or a HuggingFace repo_id.
+
+        :param inplace: If True, local paths will not be moved into
+         the models directory, but registered in place (the default).
+
+        :param variant: For HuggingFace models, this optional parameter
+         specifies which variant to download (e.g. 'fp16')
+
+        :param subfolder: When downloading HF repo_ids this can be used to
+         specify a subfolder of the HF repository to download from.
+
+        :param config: Optional dict. Any fields in this dict
+         will override corresponding autoassigned probe fields in the
+         model's config record. Use it to override
+         `name`, `description`, `base_type`, `model_type`, `format`,
+         `prediction_type`, `image_size`, and/or `ztsnr_training`.
+
+        :param access_token: Access token for use in downloading remote
+         models.
+
+        This will download the model located at `source`,
+        probe it, and install it into the models directory.
+        This call is executed asynchronously in a separate
+        thread and will issue the following events on the event bus:
+
+             - model_install_started
+             - model_install_error
+             - model_install_completed
+
+        The `inplace` flag does not affect the behavior of downloaded
+        models, which are always moved into the `models` directory.
+
+        The call returns a ModelInstallJob object which can be
+        polled to learn the current status and/or error message.
+
+        Variants recognized by HuggingFace currently are:
+        1. onnx
+        2. openvino
+        3. fp16
+        4. None (usually returns fp32 model)
+
+        """
+
+    @abstractmethod
+    def get_job(self, source: ModelSource) -> ModelInstallJob:
+        """Return the ModelInstallJob corresponding to the provided source."""
+
+    @abstractmethod
+    def list_jobs(self, source: Optional[ModelSource] = None) -> List[ModelInstallJob]:  # noqa D102
+        """
+        List active and complete install jobs.
+
+        :param source: Filter by jobs whose sources are a partial match to the argument.
```

Comment:
Find all downloads from `https://invoke.ai`? I thought it might be a useful filter feature for the frontend, but of course the frontend can do that too.

Let me know if you think this should be disabled.

Reply:
I think it's save to remove this and let the frontend do any filtering, because we don't expect this list to be too large.

Maybe at some point it'd be nice to list install jobs by type (eg HF, CivitAI, invoke.ai), but that's something to add later if the need presents itself.